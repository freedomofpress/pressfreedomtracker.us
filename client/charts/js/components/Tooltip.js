import React from 'react'
import ReactDOM from 'react-dom'

export function Tooltip({ content, x, y }) {
  const offsetX = 10
  const offletY = 60
  return ReactDOM.createPortal(
    <div
      style={{
        position: 'fixed',
        top: `${y - offletY}px`,
        left: `${x + offsetX}px`,
        backgroundColor: 'white',
        border: '5px solid black',
        pointerEvents: 'none',
        padding: 12,
      }}
    >
      {content}
    </div>,
    document.body
  )
}
