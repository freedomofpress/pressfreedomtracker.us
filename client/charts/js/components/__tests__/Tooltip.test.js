import React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import Tooltip from '../Tooltip'

jest.mock('react-dom', () => ({
	...jest.requireActual('react-dom'),
	createPortal: node => node
}))

test('renders Tooltip with mocked data', () => {
	expect(ReactTestRenderer.create(
		<Tooltip content="test" x={20} y={20} />
	)).toMatchSnapshot();
});
