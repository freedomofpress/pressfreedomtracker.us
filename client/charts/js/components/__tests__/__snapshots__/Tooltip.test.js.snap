// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tooltip with mocked data 1`] = `
<div
  style={
    {
      "backgroundColor": "white",
      "border": "5px solid black",
      "left": "30px",
      "padding": 12,
      "pointerEvents": "none",
      "position": "fixed",
      "top": "-40px",
    }
  }
>
  test
</div>
`;
